components/searchTodo.js

```diff
@@ -0,0 +1,28 @@
+import { searchTodoInput } from "./domElements.js";
+import { globalTodos, setTodos } from "../entry.js";
+import { renderTodoList } from "./domActions.js";
+
+function searchTodo() {
+  const searchedText = searchTodoInput.value.toLowerCase();
+
+  const searchedTodos = globalTodos.filter((todo) =>
+    todo.title.toLowerCase().includes(searchedText)
+  );
+
+  setTodos(searchedTodos);
+  renderTodoList();
+}
+
+export function debounceSearch(fn, delay) {
+  let timer;
+
+  return function () {
+    timer && clearTimeout(timer);
```

Comment:
This expression doesn't evaluate to anything as it doesn't have any assignment.
```suggestion
    clearTimeout(timer);
```